src/crush/CrushWrapper.cc

```diff
@@ -1702,6 +1702,10 @@ int CrushWrapper::add_bucket(
 
 int CrushWrapper::bucket_add_item(crush_bucket *bucket, int item, int weight)
 {
+  int r = crush_bucket_add_item(crush, bucket, item, weight);
+  if (r < 0) {
+    return r;
+  }
   __u32 new_size = bucket->size + 1;
```

Comment:
`crush_bucket_add_item` will have done bucket-size adjustment by now, so this is mis-counting here